src/cogs/help.py

```diff
@@ -0,0 +1,234 @@
+            description_string += (
+                f"`{item}`{dash}{alias_string}:\n"
+                f'{doc_item["doc_string"]}\n'
+                f"```{prefix}{item} {usage}```\n\n"
+            )
+    return description_string
+
+
+class Help(commands.Cog):
+    """Defines everything related towards the help command."""
+
+    def __init__(self, bot):
+        self.embed_color = discord.Color(0x2F3136)
+        self.bot = bot
+        self.config = configparser.ConfigParser()
+        self.config.read("../config.ini")
+        self.prefix = self.config.get("Bot", "Prefix")
+
+    @commands.command(name="contrib")
+    async def send_contribution_info(self, ctx):
+        """Contains information about how to contribute to the bot."""
+
+        contributing_md_url = (
+            "https://github.com/pcparadise/discordbot/blob/main/CONTRIBUTING.md"
+        )
+
+        embed = discord.Embed(
+            title="How To Contribute:",
+            description=f"Visit: [{contributing_md_url[8:]}]({contributing_md_url})",
+            color=self.embed_color,
+        )
+
+        await ctx.send(embed=embed)
+
+    @commands.command(name="dump_help_json", enabled=False)
+    async def dump_help_json(self, ctx):
+        """Dumps all help data into a parsable json format. Useful for debugging."""
+        cog_dict = cog_values(self)
+        await ctx.send("```json\n{}```".format(cog_dict))
+
+    # replaces the default help command
+    @commands.command(name="help", aliases=["h"], usage=f"[command_name|cog_name]")
+    async def send_help(self, ctx, *args):
+        """Provides information about available commands."""
+
+        argument = " ".join(args).lower().strip()
+
+        cog_dict = cog_values(self)
+
+        arg_in_misc = find_ci(cog_dict["misc_commands"], argument)
+        arg_in_dict = find_ci(cog_dict["all_commands"], argument)
+
+        if not args:  # builds the embed if no arguments where provided
+            embed = discord.Embed(
+                title="Commands and modules",
+                description=(
+                    f"View a specific module or commands info with:\n"
+                    f"```{self.prefix}help [module_name|command_name]```"
+                ),
+                color=self.embed_color,
+            )
+
+            for cog in cog_dict:
+                with suppress(Exception):  # forced to use this because pylint
+                    embed.add_field(
+                        name=cog,
+                        value=f'`{cog_dict[cog]["command_count"]} commands`',
+                        inline=True,
+                    )
+
+            if cog_dict["misc_commands"]:
+                embed.add_field(
+                    name="Uncategorized Commands:",
+                    value=misc_command_str(cog_dict),
+                    inline=False,
+                )
+
+            await ctx.send(embed=embed)
+        elif find_ci(cog_dict, argument):
```

Comment:
Rather then having else ifs - it's probably cleaner and easier to understand to just have them all be ifs and do early returns. E.G.
```python
if not args:
    #code here
    return
if find_ci(cog_dict, argument):
    #code here
    return
if arg_in_dict or arg_in_misc:
    #code here
    return
# put the code for when it matches none here e.g. the "else"
```

Reply:
If you do go this route - you can also probably simplify your code even more, namely because you could have code that run before each of the ifs as a way to initialize it's condition.